alpha-blog/app/views/articles/_article.html.erb

```diff
@@ -4,10 +4,15 @@
       <div class="col-8 mt-4">
         <div class="card text-center shadow mb-5 bg-white rounded">
           <div class="card-header font-italic">
-            by <%= link_to article&.user&.username, user_path(article.user) %>
+            by <%= link_to article&.user&.username, user_path(article&.user), class: "no-underline" %>
```

Comment:
Question. What happens when you do `user_path(nil)`

Reply:
![image](https://user-images.githubusercontent.com/67599130/122050764-62601b00-ce16-11eb-8c13-e53c6a10662a.png)
No ID will be passed in as params. So will have error when rendering the views

Reply:
Then, is there any difference for the following codes:

```
user_path(article.user)
```
vs
```
user_path(article&.user)
```

Reply:
`user_path(article&.user)`  uses the Safe Navigation Operator (&.). Lets me call methods on objects without worrying that the object may be nil(Avoiding an undefined method for `nil:NilClass` error). So if without `&.`  and if `.user` is nil then the code will not continue running.

Reply:
ok.
So, you are saying:
Given `article.user`. You use `article&.user` so that if `article` is nil, it will be `nil` instead of having an exception there.
However, `user_path(nil)` is acceptable to have an exception?

Reply:
when `article` is nil `.user` is not invoked and the result is `nil`. In that case, the arguments of `.user` are not evaluated

```ruby
# Call method `.user` on `article` only if `article` is not `nil`
user_path(article&.user)

# Equivalent to
unless article.nil?
  article.user
end

# or
if not article.nil?
  article.user
end
```

Reply:
wtf you talkin about?